test/integration/entry-integration.js

```diff
@@ -480,48 +474,76 @@ describe('Entry Api', () => {
       },
     }
 
-    before(async () => {
-      // The default space has xspace enabled in it through the feature flags
-      xSpaceEnabledContentType = await xSpaceEnabledEnvironment.getContentType(
-        TestDefaults.contentType.withCrossSpaceReferenceId
-      )
+    describe('When cross-space references feature is enabled for the space', () => {
+      let xSpaceEnabledSpace
+      let xSpaceEnabledEnvironment
+      let xSpaceEnabledContentType
+
+      before(async () => {
+        // The default space has x-space enabled in it through the feature flags
+        xSpaceEnabledSpace = await getDefaultSpace()
+        xSpaceEnabledEnvironment = await xSpaceEnabledSpace.getEnvironment('master')
+        xSpaceEnabledContentType = await xSpaceEnabledEnvironment.getContentType(
+          TestDefaults.contentType.withCrossSpaceReferenceId
+        )
+      })
 
-      // Creating a new space that doesn't have the xspace feature enabled
-      xSpaceDisabledSpace = await createTestSpace(initClient(), 'XSpaceDisabledEntry')
-      xSpaceDisabledEnvironment = await createTestEnvironment(
-        xSpaceDisabledSpace,
-        'Testing Environment'
-      )
-      xSpaceDisabledContentType = await xSpaceDisabledEnvironment.createContentTypeWithId(
-        generateRandomId('test-content-type'),
-        contentTypeData
-      )
-      await xSpaceDisabledContentType.publish()
+      test('can create, publish, unpublish and delete an entry with xspace references', async () => {
+        return xSpaceEnabledEnvironment
+          .createEntry(xSpaceEnabledContentType.sys.id, entryData)
```

Comment:
Please do not mutate the default space here! 
If you need to mutate, clone the environment using `createTestEnvironment` and do any write operation exclusively on this environment.

Reply:
Just to explain, I needed a space with cross space feature enabled and enabling it for a space needs additional steps like adding the `space_id` in the feature flags dedicated for the feature. Your idea of cloning master worked for me so I am pushing the newest changes.

Reply:
Thanks for the explanation @Cyberxon 🙏 
I think that's all totally fine, and the latest changes reflect the best possible test setup for it :)